### ISSUE TITLE
Crash when debugging in Android Emulator with architecture ARM

### ISSUE THREAD

**Issue by Happypig375** [fsprojects/Fabulous#492]
# #385 redux

<p>
1. Create a Fabulous app as usual and open the solution<br/>
2. Switch to the Android project<br/>
3-6. Create an armeabi-v7a emulator<br/>
7. <a href="https://stackoverflow.com/questions/39644144/android-emulator-not-showing-in-visual-studio">Restart Adb server to make new emulator show up</a><br/>
8. Debug the project with F5<br/>
9. Accept the recommendation for x86 dialog<br/>
10. Wait for initial load<br/>
11. Fix <code>Emulator:glTexImage2D: got err pre :(</code> errors with<br/>
<a href="https://stackoverflow.com/questions/50595704/glteximage2d-got-err-pre-0x506-internal-0x1908-format-0x1908-type-0x1401">&nbsp;&nbsp;&nbsp;&nbsp;11.1. More>Settings>Advanced>OpenGL ES Renderer: SwiftShader<br/>
&nbsp;&nbsp;&nbsp;&nbsp;11.2. More>Settings>Advanced>OpenGL ES API level: Renderer Maximum<br/></a>
12. Restart emulator, re-debug<br/>
13. Wait<br/>
14. Open Debug log
</p>

<details>
<summary>Screenshots</summary>

1.
![image](https://user-images.githubusercontent.com/19922066/60648730-1909a180-9e73-11e9-8c46-aab20a82620e.png)
2. (Visual Studio 2019 is opened)
![image](https://user-images.githubusercontent.com/19922066/60648292-3b4eef80-9e72-11e9-93b1-e386b2ef2ac6.png)
3.
![image](https://user-images.githubusercontent.com/19922066/60648254-21151180-9e72-11e9-808d-9faa5e509e0e.png)
4.
![image](https://user-images.githubusercontent.com/19922066/60647902-5c631080-9e71-11e9-855a-beb0a8ec106b.png)
5.
![image](https://user-images.githubusercontent.com/19922066/60647950-7a307580-9e71-11e9-8c2f-7f6a51b089d9.png)
6.
![image](https://user-images.githubusercontent.com/19922066/60648835-5cfca680-9e73-11e9-824c-66dfcf0440b0.png)
7.
![image](https://user-images.githubusercontent.com/19922066/60678597-9b1bb980-9eb7-11e9-939b-220f82993f3b.png)
9. 
![image](https://user-images.githubusercontent.com/19922066/60648982-ac42d700-9e73-11e9-93a9-22dfbd54a843.png)
<br/>
11.
![image](https://user-images.githubusercontent.com/19922066/60650889-9f27e700-9e77-11e9-8b6b-ba64359a5af8.png)
![image](https://user-images.githubusercontent.com/19922066/60650913-aa7b1280-9e77-11e9-8dc3-08080034ca91.png)
</details>
12.
<details>
<summary>Build log to reference while waiting</summary>

```
1>Waiting for runtime checks to complete
Detecting installed packages
Detecting installed packages...
Removing old runtime
Removing old runtime: Mono.Android.DebugRuntime...
Installing shared runtime
Target device is armeabi-v7a.
Installing the Mono shared runtime (debug - 1560182289)...
Installing platform framework
Installing the API 28 platform framework..
Runtime checks completed
1>Starting deployment to arm_v7a ...
1>Deploying to arm_v7a ...
1>Build started.
1>Project "ARMNullRef.Android.fsproj" (Install target(s)):
1>Project "ARMNullRef.fsproj" (GetTargetFrameworks target(s)):
1>Done building project "ARMNullRef.fsproj".
1>Project "ARMNullRef.fsproj" (GetTargetPath target(s)):
1>Done building project "ARMNullRef.fsproj".
1>No way to resolve conflict between "mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" and "mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e". Choosing "mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" arbitrarily.
1>C:\Program Files\Android\jdk\microsoft_dist_openjdk_1.8.0.25\bin\keytool.exe -list -alias androiddebugkey -storepass android -keypass android -keystore "C:\Users\hadri\AppData\Local\Xamarin\Mono for Android\debug.keystore" 
1>C:\Program Files (x86)\Android\android-sdk\build-tools\28.0.3\zipalign.exe -p 4 "C:\Users\hadri\source\repos\ARMNullRef\ARMNullRef.Android\obj\Debug\android\bin\com.companyname.apk" "bin\Debug\\com.companyname-Signed.apk" 
1>C:\Program Files\Android\jdk\microsoft_dist_openjdk_1.8.0.25\bin\java.exe -jar "C:\Program Files (x86)\Android\android-sdk\build-tools\28.0.3\lib\apksigner.jar" sign --ks "C:\Users\hadri\AppData\Local\Xamarin\Mono for Android\debug.keystore" --ks-pass pass:android --ks-key-alias androiddebugkey --key-pass pass:android --min-sdk-version 21 --max-sdk-version 28  C:\Users\hadri\source\repos\ARMNullRef\ARMNullRef.Android\bin\Debug\com.companyname-Signed.apk 
1>Done building project "ARMNullRef.Android.fsproj".
1>Build succeeded.
1>Deployment was successful to arm_v7a.
========== Deploy: 1 succeeded, 0 failed, 0 skipped ==========
```
</details>

14.
<details>
<summary>Debug log</summary>

```
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/ARMNullRef.dll
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Fabulous.Core.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Fabulous.CustomControls.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Fabulous.LiveUpdate.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/FormsViewGroup.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/FSharp.Core.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Newtonsoft.Json.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Arch.Core.Common.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Arch.Core.Runtime.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Arch.Lifecycle.Common.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Arch.Lifecycle.LiveData.Core.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Arch.Lifecycle.Runtime.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Arch.Lifecycle.ViewModel.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.FSharp.ResourceProvider.Runtime.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.Animated.Vector.Drawable.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.Annotations.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.AsyncLayoutInflater.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.Collections.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.Compat.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.CoordinaterLayout.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.Core.UI.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.Core.Utils.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.CursorAdapter.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.CustomTabs.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.CustomView.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.Design.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.DocumentFile.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.DrawerLayout.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.Fragment.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.Interpolator.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.Loader.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.LocalBroadcastManager.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.Media.Compat.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.Print.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.SlidingPaneLayout.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.SwipeRefreshLayout.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.Transition.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.v4.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.v7.AppCompat.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.v7.CardView.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.v7.MediaRouter.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.v7.Palette.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.v7.RecyclerView.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.Vector.Drawable.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.VersionedParcelable.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Android.Support.ViewPager.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Essentials.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Forms.Core.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Forms.Platform.Android.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Forms.Platform.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/Xamarin.Forms.Xaml.dll [External]
Loaded assembly: Mono.Android.dll [External]
Loaded assembly: Java.Interop.dll [External]
Loaded assembly: System.dll [External]
Loaded assembly: Mono.Security.dll [External]
Loaded assembly: System.Core.dll [External]
Loaded assembly: /storage/emulated/0/Android/data/com.companyname/files/.__override__/ARMNullRef.Android.dll
Loaded assembly: netstandard.dll [External]
Loaded assembly: System.Runtime.dll [External]
Loaded assembly: System.Diagnostics.Debug.dll [External]
Loaded assembly: System.Resources.ResourceManager.dll [External]
Loaded assembly: System.Globalization.dll [External]
Loaded assembly: System.Collections.Concurrent.dll [External]
Loaded assembly: System.Collections.dll [External]
Loaded assembly: System.Reflection.dll [External]
Loaded assembly: System.Reflection.Extensions.dll [External]
Loaded assembly: System.Runtime.Extensions.dll [External]
Loaded assembly: System.Linq.dll [External]

=================================================================
	Native Crash Reporting
=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Basic Fault Adddress Reporting
=================================================================
Memory around native instruction pointer (0x9391ffd0):0x9391ffc0  50 8d 2e 96 5c 8d 2e 96 68 8d 2e 96 74 8d 2e 96  P...\...h...t...
0x9391ffd0  18 ea e6 a4 1b 00 00 00 00 bf ce af 28 4b 29 96  ............(K).
0x9391ffe0  b8 1f 0a a5 13 04 00 00 00 01 10 00 00 00 00 00  ................
0x9391fff0  00 00 00 00 
00 80 2e 96 0c 80 2e 96 18 80 2e 96  ................

No native Android stacktrace (see debuggerd output).

=================================================================
	Managed Stacktrace:
=================================================================
=================================================================
```

</details>

**Comment by Happypig375** [comment]
Please look into this. The development experience is awfully degraded while this issue persists.

**Comment by TimLariviere** [comment]
I'm looking into this.
Thanks for giving the tips for making the emulator work, really useful. 😄 

I can say that on macOS it's not crashing, the app runs fine on an `Nougat 7.1 - API 25 / armebi-v7a` emulator.
Currently testing on Windows.

**Comment by TimLariviere** [comment]
I'm not reproducing :/
The emulator is too slow for the debugger of Visual Studio 2019 (Windows), the debug times out and the app is killed before starting...

But the app is deployed, and without a debugger, the app runs correctly.

**Comment by Happypig375** [comment]
I forgot to add that for the "not responding" popups, "Wait" should be pressed. I'll try reproing once again.

**Comment by TimLariviere** [comment]
I did tap the Wait button in the Not responding popups.
But the debugger of Visual Studio successfully deploys but when attaching, it times out without a "Not responding" popup and runs a `emulator force-stop` command, killing the app before being able to run it

**Comment by Happypig375** [comment]
I'm also reproing the `force-stop` command after updating to VS16.1.4. Argh.

**Comment by Happypig375** [comment]
But the C# Xamarin.Forms template is ok

**Comment by Happypig375** [comment]
Alright. Here is repro steps v3.

1. Create a blank C# Xamarin.Forms app.
2. Run it on the Android armeabi-v7a emulator with steps as stated above.
3. App runs normally.
4. Update Xamarin.Forms dependencies.
5. Install Fabulous.XamarinForms on the cross-platform project.
6. Run it again on the emulator.
7. App still runs fine.
8. 
```diff
using System;
using Xamarin.Forms;
using Xamarin.Forms.Xaml;
+using Microsoft.FSharp.Core;
+using Fabulous;
+using Fabulous.XamarinForms;

namespace CSharpReference
{
    public partial class App : Application
    {
        public App()
        {
            InitializeComponent();

            MainPage = new MainPage();

+            XamarinFormsProgramModule.run(this,
+                ProgramModule.mkSimple(
+                    FuncConvert.FromFunc(() => 0),
+                    FuncConvert.FromFunc<int, int, int>((msg, model) => model + msg),
+                    FuncConvert.FromFunc<int, FSharpFunc<int, Unit>, ViewElement>(
+                        (model, dispatch) =>
+                        Fabulous.XamarinForms.View.Button(
+                            FSharpOption<string>.Some(model.ToString()),
+                            FSharpOption<FSharpFunc<Unit, Unit>>.Some(
+                                FuncConvert.FromAction(() => dispatch.Invoke(1))),
+                            null, null, null, null, null, null, null, null, null,
+                            null, null, null, null, null, null, null, null, null, null,
+                            null, null, null, null, null, null, null, null, null, null,
+                            null, null, null, null, null, null, null, null, null, null,
+                            null, null, null, null, null, null, null, null, null, null,
+                            null, null, null, null, null, null, null, null, null, null,
+                            null, null, null, null, null)
+                    )));
        }

        protected override void OnStart()
        {
            // Handle when your app starts
        }

        protected override void OnSleep()
        {
            // Handle when your app sleeps
        }

        protected override void OnResume()
        {
            // Handle when your app resumes
        }
    }
}
```
9. Run it on the emulator.
10. 
```
=================================================================
	Native Crash Reporting
=================================================================
Got a SIGSEGV while executing native code. This usually indicates
a fatal error in the mono runtime or one of the native libraries 
used by your application.
=================================================================

=================================================================
	Basic Fault Adddress Reporting
=================================================================
Memory around native instruction pointer (0x88adefb8):0x88adefa8  00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
0x88adefb8  50 c4 41 9d 7a 00 00 00 00 4f e7 a4 38 dc 7a 8a  P.A.z....O..8.z.
0x88adefc8  b0 63 ab 88 00 00 00 00 00 01 04 00 00 00 00 00  .c..............
0x88adefd8  00 00 00 00 00 00 44 8b 0c 00 44 8b 18 00 44 8b  ......
D...D...D.

No native Android stacktrace (see debuggerd output).

=================================================================
	Managed Stacktrace:
=================================================================
=================================================================
```

**Comment by Happypig375** [comment]
As such, this issue is not related to the templates. Tag can be removed.

**Comment by SergejDK** [comment]
@Happypig375 
I hope good news for you: your initial problem with debugging shouldn't exist anylonger. I set up a new `SqueakyApp` , deployed and run it in debug on a armeabi-v7a device.
I use VS2019 16.1.6 - the newest version.

> Alright. Here is repro steps v3.

Will test this, too.

**Comment by SergejDK** [comment]
This is the Stacktrace vor the v3 repo.

```
=================================================================
	Managed Stacktrace:
=================================================================
=================================================================
07-16 18:50:33.835 D/Mono    (21265): Image addref System.Reflection[0x91c47e60] (asmctx DEFAULT) -> System.Reflection.dll[0x91683000]: 2
07-16 18:50:33.835 D/Mono    (21265): Prepared to set up assembly 'System.Reflection' (System.Reflection.dll)
07-16 18:50:33.835 D/Mono    (21265): Assembly System.Reflection[0x91c47e60] added to domain RootDomain, ref_count=1
07-16 18:50:33.842 D/Mono    (21265): AOT: image 'System.Reflection.dll.so' not found: dlopen failed: library "System.Reflection.dll.so" not found
07-16 18:50:33.846 D/Mono    (21265): AOT: image '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/lib/mono/aot-cache/arm/System.Reflection.dll.so' not found: dlopen failed: library "/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/lib/mono/aot-cache/arm/System.Reflection.dll.so" not found
07-16 18:50:33.846 D/Mono    (21265): Config attempting to parse: 'System.Reflection.dll.config'.
07-16 18:50:33.846 D/Mono    (21265): Config attempting to parse: '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/etc/mono/assemblies/System.Reflection/System.Reflection.config'.
07-16 18:50:33.846 D/Mono    (21265): Assembly Ref addref FSharp.Core[0xa30fe6e0] -> System.Reflection[0x91c47e60]: 2
07-16 18:50:33.847 D/Mono    (21265): Loading reference 0 of System.Reflection.dll asmctx DEFAULT, looking for mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
07-16 18:50:33.847 D/Mono    (21265): Assembly Ref addref System.Reflection[0x91c47e60] -> mscorlib[0xa30fd900]: 54
07-16 18:50:33.848 D/Mono    (21265): Loading reference 16 of /storage/emulated/0/Android/data/com.companyname/files/.__override__/FSharp.Core.dll asmctx DEFAULT, looking for System.Reflection.Extensions, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
07-16 18:50:33.855 D/Mono    (21265): Image addref System.Reflection.Extensions[0x91c47f80] (asmctx DEFAULT) -> System.Reflection.Extensions.dll[0x91683500]: 2
07-16 18:50:33.855 D/Mono    (21265): Prepared to set up assembly 'System.Reflection.Extensions' (System.Reflection.Extensions.dll)
07-16 18:50:33.855 D/Mono    (21265): Assembly System.Reflection.Extensions[0x91c47f80] added to domain RootDomain, ref_count=1
07-16 18:50:33.866 D/Mono    (21265): AOT: image 'System.Reflection.Extensions.dll.so' not found: dlopen failed: library "System.Reflection.Extensions.dll.so" not found
07-16 18:50:33.869 D/Mono    (21265): AOT: image '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/lib/mono/aot-cache/arm/System.Reflection.Extensions.dll.so' not found: dlopen failed: library "/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/lib/mono/aot-cache/arm/System.Reflection.Extensions.dll.so" not found
07-16 18:50:33.870 D/Mono    (21265): Config attempting to parse: 'System.Reflection.Extensions.dll.config'.
07-16 18:50:33.870 D/Mono    (21265): Config attempting to parse: '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/etc/mono/assemblies/System.Reflection.Extensions/System.Reflection.Extensions.config'.
07-16 18:50:33.870 D/Mono    (21265): Assembly Ref addref FSharp.Core[0xa30fe6e0] -> System.Reflection.Extensions[0x91c47f80]: 2
07-16 18:50:33.870 D/Mono    (21265): Loading reference 0 of System.Reflection.Extensions.dll asmctx DEFAULT, looking for mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
07-16 18:50:33.871 D/Mono    (21265): Assembly Ref addref System.Reflection.Extensions[0x91c47f80] -> mscorlib[0xa30fd900]: 55
07-16 18:50:33.873 D/Mono    (21265): Loading reference 1 of /storage/emulated/0/Android/data/com.companyname/files/.__override__/FSharp.Core.dll asmctx DEFAULT, looking for System.Linq, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
07-16 18:50:33.894 D/Mono    (21265): Image addref System.Linq[0x91c47fe0] (asmctx DEFAULT) -> System.Linq.dll[0x91684900]: 2
07-16 18:50:33.895 D/Mono    (21265): Prepared to set up assembly 'System.Linq' (System.Linq.dll)
07-16 18:50:33.895 D/Mono    (21265): Assembly System.Linq[0x91c47fe0] added to domain RootDomain, ref_count=1
07-16 18:50:33.901 D/Mono    (21265): AOT: image 'System.Linq.dll.so' not found: dlopen failed: library "System.Linq.dll.so" not found
07-16 18:50:33.905 D/Mono    (21265): AOT: image '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/lib/mono/aot-cache/arm/System.Linq.dll.so' not found: dlopen failed: library "/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/lib/mono/aot-cache/arm/System.Linq.dll.so" not found
07-16 18:50:33.905 D/Mono    (21265): Config attempting to parse: 'System.Linq.dll.config'.
07-16 18:50:33.906 D/Mono    (21265): Config attempting to parse: '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/etc/mono/assemblies/System.Linq/System.Linq.config'.
07-16 18:50:33.906 D/Mono    (21265): Assembly Ref addref FSharp.Core[0xa30fe6e0] -> System.Linq[0x91c47fe0]: 2
07-16 18:50:33.906 D/Mono    (21265): Loading reference 1 of System.Linq.dll asmctx DEFAULT, looking for System.Core, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
07-16 18:50:33.906 D/Mono    (21265): Assembly Ref addref System.Linq[0x91c47fe0] -> System.Core[0xadca3380]: 6
07-16 18:50:34.085 D/Mono    (21265): Loading reference 2 of /storage/emulated/0/Android/data/com.companyname/files/.__override__/Fabulous.dll asmctx DEFAULT, looking for System.Runtime, Version=4.1.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
07-16 18:50:34.091 D/Mono    (21265): Unloading image System.Runtime.dll [0x91684e00].
07-16 18:50:34.092 D/Mono    (21265): Image addref System.Runtime[0x91c48160] (asmctx DEFAULT) -> System.Runtime.dll[0x91bd3d00]: 5
07-16 18:50:34.092 D/Mono    (21265): Config attempting to parse: 'System.Runtime.dll.config'.
07-16 18:50:34.092 D/Mono    (21265): Config attempting to parse: '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/etc/mono/assemblies/System.Runtime/System.Runtime.config'.
07-16 18:50:34.093 D/Mono    (21265): Assembly Ref addref Fabulous[0xa30fe260] -> System.Runtime[0x91c468a0]: 3
07-16 18:50:34.320 D/Mono    (21265): Loading reference 4 of /storage/emulated/0/Android/data/com.companyname/files/.__override__/Fabulous.XamarinForms.dll asmctx DEFAULT, looking for System.Runtime, Version=4.1.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
07-16 18:50:34.326 D/Mono    (21265): Unloading image System.Runtime.dll [0x91684e00].
07-16 18:50:34.326 D/Mono    (21265): Image addref System.Runtime[0x91c48160] (asmctx DEFAULT) -> System.Runtime.dll[0x91bd3d00]: 6
07-16 18:50:34.326 D/Mono    (21265): Config attempting to parse: 'System.Runtime.dll.config'.
07-16 18:50:34.327 D/Mono    (21265): Config attempting to parse: '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/etc/mono/assemblies/System.Runtime/System.Runtime.config'.
07-16 18:50:34.327 D/Mono    (21265): Assembly Ref addref Fabulous.XamarinForms[0xa30fe3e0] -> System.Runtime[0x91c468a0]: 4
07-16 18:50:35.812 D/Mono    (21265): Loading reference 7 of /storage/emulated/0/Android/data/com.companyname/files/.__override__/FSharp.Core.dll asmctx DEFAULT, looking for System.Collections, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
07-16 18:50:35.819 D/Mono    (21265): Image addref System.Collections[0x91c48220] (asmctx DEFAULT) -> System.Collections.dll[0x91685300]: 2
07-16 18:50:35.819 D/Mono    (21265): Prepared to set up assembly 'System.Collections' (System.Collections.dll)
07-16 18:50:35.819 D/Mono    (21265): Assembly System.Collections[0x91c48220] added to domain RootDomain, ref_count=1
07-16 18:50:35.826 D/Mono    (21265): AOT: image 'System.Collections.dll.so' not found: dlopen failed: library "System.Collections.dll.so" not found
07-16 18:50:35.829 D/Mono    (21265): AOT: image '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/lib/mono/aot-cache/arm/System.Collections.dll.so' not found: dlopen failed: library "/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/lib/mono/aot-cache/arm/System.Collections.dll.so" not found
07-16 18:50:35.830 D/Mono    (21265): Config attempting to parse: 'System.Collections.dll.config'.
07-16 18:50:35.830 D/Mono    (21265): Config attempting to parse: '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/etc/mono/assemblies/System.Collections/System.Collections.config'.
07-16 18:50:35.830 D/Mono    (21265): Assembly Ref addref FSharp.Core[0xa30fe6e0] -> System.Collections[0x91c48220]: 2
07-16 18:50:35.830 D/Mono    (21265): Loading reference 0 of System.Collections.dll asmctx DEFAULT, looking for mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
07-16 18:50:35.830 D/Mono    (21265): Assembly Ref addref System.Collections[0x91c48220] -> mscorlib[0xa30fd900]: 56
07-16 18:50:35.983 D/Mono    (21265): Loading reference 10 of /storage/emulated/0/Android/data/com.companyname/files/.__override__/FSharp.Core.dll asmctx DEFAULT, looking for System.Resources.ResourceManager, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
07-16 18:50:35.990 D/Mono    (21265): Image addref System.Resources.ResourceManager[0x91c48160] (asmctx DEFAULT) -> System.Resources.ResourceManager.dll[0x91684e00]: 2
07-16 18:50:35.991 D/Mono    (21265): Prepared to set up assembly 'System.Resources.ResourceManager' (System.Resources.ResourceManager.dll)
07-16 18:50:35.991 D/Mono    (21265): Assembly System.Resources.ResourceManager[0x91c48160] added to domain RootDomain, ref_count=1
07-16 18:50:35.999 D/Mono    (21265): AOT: image 'System.Resources.ResourceManager.dll.so' not found: dlopen failed: library "System.Resources.ResourceManager.dll.so" not found
07-16 18:50:36.002 D/Mono    (21265): AOT: image '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/lib/mono/aot-cache/arm/System.Resources.ResourceManager.dll.so' not found: dlopen failed: library "/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/lib/mono/aot-cache/arm/System.Resources.ResourceManager.dll.so" not found
07-16 18:50:36.003 D/Mono    (21265): Config attempting to parse: 'System.Resources.ResourceManager.dll.config'.
07-16 18:50:36.004 D/Mono    (21265): Config attempting to parse: '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/etc/mono/assemblies/System.Resources.ResourceManager/System.Resources.ResourceManager.config'.
07-16 18:50:36.004 D/Mono    (21265): Assembly Ref addref FSharp.Core[0xa30fe6e0] -> System.Resources.ResourceManager[0x91c48160]: 2
07-16 18:50:36.004 D/Mono    (21265): Loading reference 0 of System.Resources.ResourceManager.dll asmctx DEFAULT, looking for mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
07-16 18:50:36.004 D/Mono    (21265): Assembly Ref addref System.Resources.ResourceManager[0x91c48160] -> mscorlib[0xa30fd900]: 57
07-16 18:50:36.005 D/Mono    (21265): Loading reference 11 of /storage/emulated/0/Android/data/com.companyname/files/.__override__/FSharp.Core.dll asmctx DEFAULT, looking for System.Globalization, Version=4.0.10.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
07-16 18:50:36.011 D/Mono    (21265): Image addref System.Globalization[0x91c48040] (asmctx DEFAULT) -> System.Globalization.dll[0x91685800]: 2
07-16 18:50:36.012 D/Mono    (21265): Prepared to set up assembly 'System.Globalization' (System.Globalization.dll)
07-16 18:50:36.012 D/Mono    (21265): Assembly System.Globalization[0x91c48040] added to domain RootDomain, ref_count=1
07-16 18:50:36.018 D/Mono    (21265): AOT: image 'System.Globalization.dll.so' not found: dlopen failed: library "System.Globalization.dll.so" not found
07-16 18:50:36.021 D/Mono    (21265): AOT: image '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/lib/mono/aot-cache/arm/System.Globalization.dll.so' not found: dlopen failed: library "/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/lib/mono/aot-cache/arm/System.Globalization.dll.so" not found
07-16 18:50:36.022 D/Mono    (21265): Config attempting to parse: 'System.Globalization.dll.config'.
07-16 18:50:36.022 D/Mono    (21265): Config attempting to parse: '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/etc/mono/assemblies/System.Globalization/System.Globalization.config'.
07-16 18:50:36.022 D/Mono    (21265): Assembly Ref addref FSharp.Core[0xa30fe6e0] -> System.Globalization[0x91c48040]: 2
07-16 18:50:36.023 D/Mono    (21265): Loading reference 0 of System.Globalization.dll asmctx DEFAULT, looking for mscorlib, Version=2.0.5.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e
07-16 18:50:36.023 D/Mono    (21265): Assembly Ref addref System.Globalization[0x91c48040] -> mscorlib[0xa30fd900]: 58
07-16 18:50:36.528 E/mono-rt (21265): /proc/self/maps:
07-16 18:50:36.528 E/mono-rt (21265): 12c00000-12e08000 rw-p 00000000 00:04 2754       /dev/ashmem/dalvik-main space (deleted)
07-16 18:50:36.528 E/mono-rt (21265): 12e08000-22c00000 ---p 00208000 00:04 2754       /dev/ashmem/dalvik-main space (deleted)
07-16 18:50:36.528 E/mono-rt (21265): 22c00000-22c01000 rw-p 00000000 00:04 2755       /dev/ashmem/dalvik-main space 1 (deleted)
07-16 18:50:36.528 E/mono-rt (21265): 22c01000-32c00000 ---p 00001000 00:04 2755       /dev/ashmem/dalvik-main space 1 (deleted)
07-16 18:50:36.528 E/mono-rt (21265): 704ef000-706f1000 rw-p 00000000 fe:20 7351       /data/dalvik-cache/arm/system@framework@boot.art
07-16 18:50:36.528 E/mono-rt (21265): 706f1000-7080c000 rw-p 00000000 fe:20 7352       /data/dalvik-cache/arm/system@framework@boot-core-libart.art
07-16 18:50:36.529 E/mono-rt (21265): 7080c000-70836000 rw-p 00000000 fe:20 7353       /data/dalvik-cache/arm/system@framework@boot-conscrypt.art
07-16 18:50:36.529 E/mono-rt (21265): 70836000-70860000 rw-p 00000000 fe:20 7354       /data/dalvik-cache/arm/system@framework@boot-okhttp.art
07-16 18:50:36.529 E/mono-rt (21265): 70860000-70862000 rw-p 00000000 fe:20 7355       /data/dalvik-cache/arm/system@framework@boot-core-junit.art
07-16 18:50:36.529 E/mono-rt (21265): 70862000-7089a000 rw-p 00000000 fe:20 7356       /data/dalvik-cache/arm/system@framework@boot-bouncycastle.art
07-16 18:50:36.529 E/mono-rt (21265): 7089a000-708d1000 rw-p 00000000 fe:20 7357       /data/dalvik-cache/arm/system@framework@boot-ext.art
07-16 18:50:36.529 E/mono-rt (21265): 708d1000-70e89000 rw-p 00000000 fe:20 7358       /data/dalvik-cache/arm/system@framework@boot-framework.art
07-16 18:50:36.529 E/mono-rt (21265): 70e89000-70f07000 rw-p 00000000 fe:20 7359       /data/dalvik-cache/arm/system@framework@boot-telephony-common.art
07-16 18:50:36.529 E/mono-rt (21265): 70f07000-70f0d000 rw-p 00000000 fe:20 7360       /data/dalvik-cache/arm/system@framework@boot-voip-common.art
07-16 18:50:36.529 E/mono-rt (21265): 70f0d000-70f18000 rw-p 00000000 fe:20 7361       /data/dalvik-cache/arm/system@framework@boot-ims-common.art
07-16 18:50:36.529 E/mono-rt (21265): 70f18000-70f30000 rw-p 00000000 fe:20 7362       /data/dalvik-cache/arm/system@framework@boot-apache-xml.art
07-16 18:50:36.529 E/mono-rt (21265): 70f30000-70f50000 rw-p 00000000 fe:20 7363       /data/dalvik-cache/arm/system@framework@boot-org.apache.http.legacy.boot.art
07-16 18:50:36.529 E/mono-rt (21265): 70f50000-7146a000 r--p 00000000 fe:00 981        /system/framework/arm/boot.oat
07-16 18:50:36.529 E/mono-rt (21265): 7146a000-71755000 r-xp 0051a000 fe:00 981        /system/framework/arm/boot.oat
07-16 18:50:36.529 E/mono-rt (21265): 71755000-71756000 r--p 00805000 fe:00 981        /system/framework/arm/boot.oat
07-16 18:50:36.530 E/mono-rt (21265): 71756000-71757000 rw-p 00806000 fe:00 981        /system/framework/arm/boot.oat
07-16 18:50:36.530 E/mono-rt (21265): 71757000-71bd4000 r--p 00000000 fe:00 965        /system/framework/arm/boot-core-libart.oat
07-16 18:50:36.530 E/mono-rt (21265): 71bd4000-71e16000 r-xp 0047d000 fe:00 965        /system/framework/arm/boot-core-libart.oat
07-16 18:50:36.530 E/mono-rt (21265): 71e16000-71e17000 r--p 006bf000 fe:00 965        /system/framework/arm/boot-core-libart.oat
07-16 18:50:36.530 E/mono-rt (21265): 71e17000-71e18000 rw-p 006c0000 fe:00 965        /system/framework/arm/boot-core-libart.oat
07-16 18:50:36.539 F/libc    (21265): Fatal signal 11 (SIGSEGV), code 2, fault addr 0x91588d78 in tid 21265 (com.companyname)
```

**Comment by TimLariviere** [comment]
Could it be an another issue with Mono ARM?
The only thing I found with this stack trace is that it seems to be a stack overflow exception.

**Comment by SergejDK** [comment]
Mono or VS. Don't know for sure.
But it could be that somewhere one function of us do some crazy stuff, which I don't believe but who knows. 
I'll try to take at the logs and maybe identify which function makes the call. With this information we can get better help from others.

**Comment by TimLariviere** [comment]
Just a quick remark on the repro code. The root element should be a page not a button, otherwise Xamarin.Forms will crash (with an explicit message usually).
Don't know if it has an impact.

**Comment by Happypig375** [comment]
The debugger should stop on the Exception from not using a Page as root, but I cannot even see that happen.

**Comment by SergejDK** [comment]
The error comes from this line:
`return FuncConvert.FromFunc(() => 0);`

Here is the stacktrace when it hits the breakpoint on the function:
```
Resolved pending breakpoint at 'App.xaml.cs:40,1' to int ARMCSharpTest.App.<>c.<getInit>b__1_0 () [0x00000].
Resolved pending breakpoint at 'App.xaml.cs:45,1' to int ARMCSharpTest.App.<>c.<getMM>b__2_0 (int msg, int model) [0x00000].
Loaded assembly: System.Diagnostics.Debug.dll [External]
07-17 12:10:55.629 I/art     (25543): Not late-enabling -Xcheck:jni (already on)
07-17 12:10:55.737 W/art     (25543): Unknown instruction set features for ARM CPU variant (generic) using conservative defaults
07-17 12:10:57.464 W/art     (25543): Attempt to remove non-JNI local reference, dumping thread
07-17 12:10:57.465 W/art     (25543): Attempt to remove non-JNI local reference, dumping thread
07-17 12:10:57.467 W/monodroid(25543): Creating public update directory: `/data/user/0/com.companyname/files/.__override__`
07-17 12:10:57.468 W/monodroid(25543): Using override path: /data/user/0/com.companyname/files/.__override__
07-17 12:10:57.496 W/monodroid(25543): Using override path: /storage/emulated/0/Android/data/com.companyname/files/.__override__
07-17 12:10:57.503 W/monodroid(25543): Using runtime path: /data/app/Mono.Android.DebugRuntime-1/lib/arm
07-17 12:10:57.503 W/monodroid(25543): checking directory: `/data/user/0/com.companyname/files/.__override__/lib`
07-17 12:10:57.503 W/monodroid(25543): directory does not exist: `/data/user/0/com.companyname/files/.__override__/lib`
07-17 12:10:57.503 W/monodroid(25543): checking directory: `/storage/emulated/0/Android/data/com.companyname/files/.__override__/lib`
07-17 12:10:57.518 W/monodroid(25543): directory does not exist: `/storage/emulated/0/Android/data/com.companyname/files/.__override__/lib`
07-17 12:10:57.518 W/monodroid(25543): checking directory: `/storage/emulated/0/../legacy/Android/data/com.companyname/files/.__override__/lib`
07-17 12:10:57.519 W/monodroid(25543): directory does not exist: `/storage/emulated/0/../legacy/Android/data/com.companyname/files/.__override__/lib`
07-17 12:10:57.519 W/monodroid(25543): Trying to load sgen from: /data/user/0/com.companyname/files/.__override__/libmonosgen-2.0.so
07-17 12:10:57.519 W/monodroid(25543): Trying to load sgen from: /storage/emulated/0/Android/data/com.companyname/files/.__override__/libmonosgen-2.0.so
07-17 12:10:57.525 W/monodroid(25543): Trying to load sgen from: /storage/emulated/0/../legacy/Android/data/com.companyname/files/.__override__/libmonosgen-2.0.so
07-17 12:10:57.526 W/monodroid(25543): Trying to load sgen from: /data/app/com.companyname-1/lib/arm/libmonosgen-2.0.so
07-17 12:10:57.526 W/monodroid(25543): Trying to load sgen from: /data/user/0/com.companyname/files/.__override__/links/libmonosgen-2.0.so
07-17 12:10:57.610 W/monodroid-debug(25543): Trying to initialize the debugger with options: --debugger-agent=transport=dt_socket,loglevel=0,address=127.0.0.1:29322,server=y,embedding=1
07-17 12:10:57.675 W/monodroid-debug(25543): Accepted stdout connection: 32
07-17 12:10:57.692 W/monodroid(25543): directory does not exist: `/data/user/0/com.companyname/files/.__override__/typemaps`
07-17 12:10:57.693 W/monodroid(25543): directory does not exist: `/storage/emulated/0/Android/data/com.companyname/files/.__override__/typemaps`
07-17 12:10:59.572 D/Mono    (25543): Image addref mscorlib[0xa30fd900] (asmctx DEFAULT) -> mscorlib.dll[0xa4fa6600]: 2
07-17 12:10:59.573 D/Mono    (25543): Prepared to set up assembly 'mscorlib' (mscorlib.dll)
07-17 12:10:59.590 D/Mono    (25543): AOT: image 'mscorlib.dll.so' not found: dlopen failed: library "mscorlib.dll.so" not found
07-17 12:10:59.596 D/Mono    (25543): AOT: image '/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/lib/mono/aot-cache/arm/mscorlib.dll.so' not found: dlopen failed: library "/Users/builder/jenkins/workspace/xamarin-android-d16-1/xamarin-android/external/mono/sdks/out/android-armeabi-v7a-release/lib/mono/aot-cache/arm/mscorlib.dll.so" not found
07-17 12:10:59.598 D/Mono    (25543): Config attempting to parse: 'mscorlib.dll.config'.
```

There it says known instruction set features for ARM CPU.... 

After the function it runs the SIGSEGV error happens.

**Comment by TimLariviere** [comment]
@SergejDK What happens if you remove Fabulous (including packages) and only keep FuncConvert.FromFunc(() => 0)?

**Comment by SergejDK** [comment]
You mean removing the nuget packages of Fabulous/Fabulous.XamarinForms

**Comment by TimLariviere** [comment]
In a first time, can you test this code and see if there is the same crash?
```csharp
using System;
using Xamarin.Forms;
using Xamarin.Forms.Xaml;
using Microsoft.FSharp.Core;

namespace CSharpReference
{
    public partial class App : Application
    {
        public App()
        {
            InitializeComponent();

            MainPage = new MainPage();
            
            var func = FuncConvert.FromFunc(() => 0);
            func.Invoke(); // Or whatever that can execute it
        }

        protected override void OnStart()
        {
            // Handle when your app starts
        }

        protected override void OnSleep()
        {
            // Handle when your app sleeps
        }

        protected override void OnResume()
        {
            // Handle when your app resumes
        }
    }
}

```

**Comment by SergejDK** [comment]
Tried but it works. Only interesting stacktrace i get:

```
07-17 14:11:09.022 I/art     ( 2773): Rejecting re-init on previously-failed class java.lang.Class<android.support.v4.view.ViewCompat$OnUnhandledKeyEventListenerWrapper>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/view/View$OnUnhandledKeyEventListener;
07-17 14:11:09.022 I/art     ( 2773):   at void android.support.v4.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, android.support.v4.view.OnApplyWindowInsetsListener) (ViewCompat.java:2203)
07-17 14:11:09.022 I/art     ( 2773):   at android.view.ViewGroup android.support.v7.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:637)
07-17 14:11:09.022 I/art     ( 2773):   at void android.support.v7.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:518)
07-17 14:11:09.022 I/art     ( 2773):   at void android.support.v7.app.AppCompatDelegateImpl.initWindowDecorActionBar() (AppCompatDelegateImpl.java:319)
07-17 14:11:09.022 I/art     ( 2773):   at android.support.v7.app.ActionBar android.support.v7.app.AppCompatDelegateImpl.getSupportActionBar() (AppCompatDelegateImpl.java:306)
07-17 14:11:09.022 I/art     ( 2773):   at void android.support.v7.app.AppCompatDelegateImpl.setSupportActionBar(android.support.v7.widget.Toolbar) (AppCompatDelegateImpl.java:343)
07-17 14:11:09.023 I/art     ( 2773):   at void android.support.v7.app.AppCompatActivity.setSupportActionBar(android.support.v7.widget.Toolbar) (AppCompatActivity.java:130)
07-17 14:11:09.023 I/art     ( 2773):   at void md5d99c5b241313b876eeb6f7c15fddc566.MainActivity.n_onCreate(android.os.Bundle) (MainActivity.java:-2)
07-17 14:11:09.023 I/art     ( 2773):   at void md5d99c5b241313b876eeb6f7c15fddc566.MainActivity.onCreate(android.os.Bundle) (MainActivity.java:30)
07-17 14:11:09.023 I/art     ( 2773):   at void android.app.Activity.performCreate(android.os.Bundle) (Activity.java:6679)
07-17 14:11:09.023 I/art     ( 2773):   at void android.app.Instrumentation.callActivityOnCreate(android.app.Activity, android.os.Bundle) (Instrumentation.java:1118)
07-17 14:11:09.023 I/art     ( 2773):   at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2618)
07-17 14:11:09.023 I/art     ( 2773):   at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2726)
07-17 14:11:09.023 I/art     ( 2773):   at void android.app.ActivityThread.-wrap12(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
07-17 14:11:09.023 I/art     ( 2773):   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1477)
07-17 14:11:09.023 I/art     ( 2773):   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:102)
07-17 14:11:09.023 I/art     ( 2773):   at void android.os.Looper.loop() (Looper.java:154)
07-17 14:11:09.023 I/art     ( 2773):   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6119)
07-17 14:11:09.023 I/art     ( 2773):   at java.lang.Object java.lang.reflect.Method.invoke!(java.lang.Object, java.lang.Object[]) (Method.java:-2)
07-17 14:11:09.023 I/art     ( 2773):   at void com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run() (ZygoteInit.java:886)
07-17 14:11:09.024 I/art     ( 2773):   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:776)
07-17 14:11:09.024 I/art     ( 2773): Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.View$OnUnhandledKeyEventListener" on path: DexPathList[[zip file "/data/app/com.companyname-1/base.apk"],nativeLibraryDirectories=[/data/app/com.companyname-1/lib/arm, /data/app/com.companyname-1/base.apk!/lib/armeabi-v7a, /system/lib, /vendor/lib]]
07-17 14:11:09.024 I/art     ( 2773):   at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:56)
07-17 14:11:09.024 I/art     ( 2773):   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:380)
07-17 14:11:09.024 I/art     ( 2773):   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
07-17 14:11:09.024 I/art     ( 2773):   at void android.support.v4.view.ViewCompat.setOnApplyWindowInsetsListener(android.view.View, android.support.v4.view.OnApplyWindowInsetsListener) (ViewCompat.java:2203)
07-17 14:11:09.024 I/art     ( 2773):   at android.view.ViewGroup android.support.v7.app.AppCompatDelegateImpl.createSubDecor() (AppCompatDelegateImpl.java:637)
07-17 14:11:09.024 I/art     ( 2773):   at void android.support.v7.app.AppCompatDelegateImpl.ensureSubDecor() (AppCompatDelegateImpl.java:518)
07-17 14:11:09.024 I/art     ( 2773):   at void android.support.v7.app.AppCompatDelegateImpl.initWindowDecorActionBar() (AppCompatDelegateImpl.java:319)
07-17 14:11:09.024 I/art     ( 2773):   at android.support.v7.app.ActionBar android.support.v7.app.AppCompatDelegateImpl.getSupportActionBar() (AppCompatDelegateImpl.java:306)
07-17 14:11:09.024 I/art     ( 2773):   at void android.support.v7.app.AppCompatDelegateImpl.setSupportActionBar(android.support.v7.widget.Toolbar) (AppCompatDelegateImpl.java:343)
07-17 14:11:09.024 I/art     ( 2773):   at void android.support.v7.app.AppCompatActivity.setSupportActionBar(android.support.v7.widget.Toolbar) (AppCompatActivity.java:130)
07-17 14:11:09.024 I/art     ( 2773):   at void md5d99c5b241313b876eeb6f7c15fddc566.MainActivity.n_onCreate(android.os.Bundle) (MainActivity.java:-2)
07-17 14:11:09.024 I/art     ( 2773):   at void md5d99c5b241313b876eeb6f7c15fddc566.MainActivity.onCreate(android.os.Bundle) (MainActivity.java:30)
07-17 14:11:09.024 I/art     ( 2773):   at void android.app.Activity.performCreate(android.os.Bundle) (Activity.java:6679)
07-17 14:11:09.024 I/art     ( 2773):   at void android.app.Instrumentation.callActivityOnCreate(android.app.Activity, android.os.Bundle) (Instrumentation.java:1118)
07-17 14:11:09.024 I/art     ( 2773):   at android.app.Activity android.app.ActivityThread.performLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent) (ActivityThread.java:2618)
07-17 14:11:09.024 I/art     ( 2773):   at void android.app.ActivityThread.handleLaunchActivity(android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:2726)
07-17 14:11:09.024 I/art     ( 2773):   at void android.app.ActivityThread.-wrap12(android.app.ActivityThread, android.app.ActivityThread$ActivityClientRecord, android.content.Intent, java.lang.String) (ActivityThread.java:-1)
07-17 14:11:09.024 I/art     ( 2773):   at void android.app.ActivityThread$H.handleMessage(android.os.Message) (ActivityThread.java:1477)
07-17 14:11:09.024 I/art     ( 2773):   at void android.os.Handler.dispatchMessage(android.os.Message) (Handler.java:102)
07-17 14:11:09.024 I/art     ( 2773):   at void android.os.Looper.loop() (Looper.java:154)
07-17 14:11:09.024 I/art     ( 2773):   at void android.app.ActivityThread.main(java.lang.String[]) (ActivityThread.java:6119)
07-17 14:11:09.024 I/art     ( 2773):   at java.lang.Object java.lang.reflect.Method.invoke!(java.lang.Object, java.lang.Object[]) (Method.java:-2)
07-17 14:11:09.024 I/art     ( 2773):   at void com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run() (ZygoteInit.java:886)
07-17 14:11:09.024 I/art     ( 2773):   at void com.android.internal.os.ZygoteInit.main(java.lang.String[]) (ZygoteInit.java:776)
```

**Comment by SergejDK** [comment]
The next steps I'll try:
1. add nuget Fabulouos
2. add nuget Fablous.XamarinForms
3. try the function in function `mkSimple`

One of these steps should produc the error. I think that it is the mkSimple but I need to test first.

**Comment by Happypig375** [comment]
This also has been fixed by VS16.2.0.

**Comment by TimLariviere** [comment]
Closing since it has apparently been fixed.
@Happypig375 Don't hesitate to put a message if it's not the case.
